lua/entities/gmod_tardis/modules/sh_redecorate.lua

```diff
@@ -0,0 +1,175 @@
+TARDIS:AddSetting({
+	id="redecorate-interior",
+	name="Redecoration interior",
+	value="default",
+	networked=true
+})
+
+local ext_saved_data_names = {
+	"cloak",
+	"floatfirst",
+	"hads",
+	"spindir",
+	"spindir_prev",
+	"spindir_before_door",
+	"handbrake",
+	"physlock",
+	"demat-fast",
+	"fastreturn-pos",
+	"fastreturn-ang"
+}
+local int_saved_data_names = {
+	"security"
+}
+
+local function apply_saved_data(object, data, names)
+	if not IsValid(object) or not data then return end
+
+	for i,name in ipairs(names) do
+		if data[name] then
+			object:SetData(name, data[name], true)
+		end
+	end
+end
+
+
+if SERVER then
+
+	ENT:AddHook("ShouldRedecorate", "redecorate_toggled", function(self)
+		return (self:GetData("redecorate",false) and self:GetData("redecorate-interior") ~= self.metadata.ID) and true or nil
+	end)
+
+	function ENT:Redecorate()
+
+		local ply = self:GetCreator()
+
+		self:SetData("repairing", false, true)
+		self:ChangeHealth(TARDIS:GetSetting("health-max"))
+		self:SetPower(true)
+
+		-- save tardis state
+		local ext_saved_data = {}
+		local int_saved_data = {}
+		for k,v in ipairs(ext_saved_data_names) do
+			ext_saved_data[v] = self:GetData(v)
+		end
+		if IsValid(self.interior) then
+			for k,v in ipairs(int_saved_data_names) do
+				int_saved_data[v] = self.interior:GetData(v)
+			end
+		end
+
+		local child = TARDIS:SpawnTARDIS(ply, {
+			pos = self:GetPos(),
+			metadataID = self:GetData("redecorate-interior") or "default",
+			redecorate_parent = self,
+			ext_data = ext_saved_data,
+			int_data = int_saved_data,
+		})
+
+		if not IsValid(child) then
+			error("Redecoration failed: failed to spawn the replacement")
+		end
+		return IsValid(child) -- if gone wrong, finish repair
+	end
+
+	ENT:AddHook("StopDemat", "redecorate_remove_parent", function(self)
+		if self:GetData("redecorate") then
+			self:GetData("redecorate_child"):SetData("redecorate_parent", nil, true)
+			self:Remove()
+		end
+	end)
+
+	ENT:AddHook("MatStart", "redecorate_sync", function(self)
+		local parent = self:GetData("redecorate_parent")
+		if not parent then return end
+
+		self:SetCollisionGroup(COLLISION_GROUP_WORLD)
+		self:SetPos(parent:GetPos())
```

Comment:
I've added safeguards to prevent that one in a25844d5f13117fd639e2c45d1f69c777d3d8622
Now, when the parent is dropped, it cannot be picked up again (COLLISION_GROUP_IN_VEHICLE)